src/utils/diagnostics.ts

```diff
@@ -313,6 +316,66 @@ const diagnoseCardKeys = async (
   return diagnostics;
 };
 
+/**
+ * Provide diagnostics related to variables within an extension.
+ * Users are warned if variables are defined but are not referenced.
+ * Errors are raised if variables are referenced that are not defined.
+ * @param document text document where diagnostics should be applied
+ * @param extension extension.yaml serialized as object
+ * @returns list of diagnostics
+ */
+const diagnoseVariables = async (
+  document: vscode.TextDocument,
+  extension: ExtensionStub,
+): Promise<vscode.Diagnostic[]> => {
+  const content = document.getText();
+  const diagnostics: vscode.Diagnostic[] = [];
+
+  // Honor the user's settings
+  if (
+    !vscode.workspace
+      .getConfiguration("dynatraceExtensions", null)
+      .get("diagnostics.variables", false) as boolean
+  ) {
+    return [];
+  }
+
+  const refVars: Array<string> = [];
+  const defVars: Array<string> = [];
+  const varRegEx = /\s(var:[a-zA-z.-_]+)\s/gm;
```

Comment:
I would adjust the regex regarding the leading and trailing `\s`.
There are 2 situations which spring to mind:
1. User quotes their string. E.g. `filter: "var:myVar"` (this happens a lot)
2. Due to weird structure, the var is at the very end of the doc (no more content after it, therefore no trailing `\s`

Reply:
Ah, both make sense